src/sentry/snuba/metrics/utils.py

```diff
@@ -430,7 +432,7 @@ def to_intervals(
 
 def get_num_intervals(
     start: Optional[datetime],
-    end: datetime,
+    end: Optional[datetime],
```

Comment:
To match the typing of the function we call
https://github.com/getsentry/sentry/blob/03f549ce7f4eeb7e565e8044f6aad387717c8237/src/sentry/snuba/metrics/utils.py#L393-L395